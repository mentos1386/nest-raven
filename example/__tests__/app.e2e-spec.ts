import { Test, TestingModule } from '@nestjs/testing';
import * as request from 'supertest';
import { INestApplication } from '@nestjs/common';
import { getApolloServer } from '@nestjs/graphql';
import type { ApolloServerBase } from 'apollo-server-core';
import gql from 'graphql-tag';
import { AppModule } from './../src/app.module';

describe('AppModule', () => {
  let app: INestApplication;
  let apolloClient: ApolloServerBase;

  beforeAll(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();

    apolloClient = getApolloServer(moduleFixture);
  });

  afterAll(() => app.close());

  it('defined', () => expect(app).toBeDefined());

  it('/ (GET)', () => request(app.getHttpServer()).get('/').expect(500));

  it('/graphql (GET)', () =>
    request(app.getHttpServer()).get('/graphql').expect(400));

  it('/graphql(POST) forbiddenError warning', async () => {
    const result = await apolloClient.executeOperation({
      query: gql`
        query {
          unauthorizedException
        }
      `,
      variables: {},
    });
    expect(result.errors).toMatchSnapshot();
  });

  it('/graphql(POST) forbiddenError', async () => {
    const result = await apolloClient.executeOperation({
      query: gql`
        query {
          forbiddenException
        }
      `,
      variables: {},
    });
    expect(result.errors).toMatchSnapshot();
  });
});
