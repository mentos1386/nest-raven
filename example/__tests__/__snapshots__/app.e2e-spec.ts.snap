// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModule /graphql(POST) forbiddenError 1`] = `
Array [
  Object {
    "extensions": Object {
      "code": "FORBIDDEN",
      "response": Object {
        "message": "Forbidden",
        "statusCode": 403,
      },
    },
    "message": "Forbidden",
  },
]
`;

exports[`AppModule /graphql(POST) forbiddenError warning 1`] = `
Array [
  Object {
    "extensions": Object {
      "code": "UNAUTHENTICATED",
      "response": Object {
        "message": "Unauthorized",
        "statusCode": 401,
      },
    },
    "message": "Unauthorized",
  },
]
`;
